jujugui/static/gui/src/app/components/constraints/constraints.js

```diff
@@ -48,21 +49,76 @@ YUI.add('constraints', function() {
       @method _handleValueChanged
     */
     _handleValueChanged: function() {
-      var refs = this.refs;
+      const refs = this.refs;
+      const defaultRef = {value: null};
+      const arch = refs.archConstraintSelect || defaultRef;
+      const cpu = refs.cpuConstraintInput || defaultRef;
+      const cores = refs.coresConstraintInput || defaultRef;
+      const mem = refs.memConstraintInput || defaultRef;
+      const disk = refs.diskConstraintInput || defaultRef;
+      const series = refs.seriesConstraintSelect || defaultRef;
       this.props.valuesChanged({
-        'cpu-power': refs.cpuConstraintInput.value,
-        cores: refs.coresConstraintInput.value,
-        mem: refs.memConstraintInput.value,
-        'root-disk': refs.diskConstraintInput.value
+        'arch': arch.value,
+        'cpu-power': cpu.value,
+        cores: cores.value,
+        mem: mem.value,
+        'root-disk': disk.value,
+        // Even if series are not technically a constraint in Juju, we include
+        // them as part of constraints. When adding the machine the series is
+        // extracted and handled separately in the API call.
+        series: series.value
       });
     },
 
     render: function() {
-      var disabled = this.props.disabled;
-      return (
-        <div className="constraints">
-          <label htmlFor="cpu-constraint"
-            className="constraints__label">
+      const props = this.props;
+      const disabled = props.disabled;
+      let series;
+      // Only allow selecting a series if there is no unit already assigned.
+      // If there is a unit, the machine must have the same series as the unit.
+      if (!props.hasUnit) {
+        // Generate a list of series options.
+        const machineSeries = window.jujulib.SERIES.filter(ser => {
```

Comment:
From the URL perspective bundle is a series. I agree we could provide the list of series.